### PR TITLE
[MM-46451] Avoiding unnecessary CloudFlare requests

### PR DESCRIPTION
**Pull request by mirshahriar**
#### Summary

We will only update DNS records when changes necessitate it.

Steps:
- Get DNS records
- If there are no existing records, create a new one.
- If record already exists, only update if record is modified.

#### Ticket Link

Resolves [this ticket](https://mattermost.atlassian.net/browse/MM-46451)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: internal/tools/cloudflare/dns.go [hunk 1]
```diff
@@ -114,42 +114,64 @@ func (c *Client) CreateDNSRecords(dnsNames []string, dnsEndpoints []string, logg
...
+		doUpdate := false
+
+		if recordToUpdate.Content != record.Content || recordToUpdate.TTL != record.TTL {
+			doUpdate = true
+		}
+		if recordToUpdate.Proxied != nil && record.Proxied != nil {
+			if *recordToUpdate.Proxied != *record.Proxied {
+				doUpdate = true
+			}
+		} else if recordToUpdate.Proxied != nil || record.Proxied != nil {
```
Comment (fmartingr) [inline internal/tools/cloudflare/dns.go:148]: Shouldn't this be: ?

```suggestion
		} else if recordToUpdate.Proxied != record.Proxied {
```

**Review by fmartingr — APPROVED** [review]
LGTM! 👍 I don't particularly like the anonymous function, but I like that all logic is still present on the same method with it.

**Comment by mirshahriar** [comment]
>  I don't particularly like the anonymous function

We require a function to address memory leaks caused by `defer` calls.  This function has no other callers. That is why I made it anonymous. But you are right.

**Review by stafot — APPROVED** [review]

**Review by gabrieljackson — APPROVED** [review]
👍

**Pull request merged**